build.rs

```diff
@@ -0,0 +1,341 @@
+            Ok(true) => {
+                // Point the patch file to our generated svd and apply the patch.
+                let includer_content = format!(
+                    r#"_svd: {}
+
+_include:
+- {}
+"#,
+                    unpatched_svd_path.display(),
+                    inputs.patch_path.display()
+                );
+                let Ok(()) = fs::write(&includer_path, &includer_content) else {
+                    println!("cargo::error=Failed to write includer file for {}!", mcu);
+                    return Err(());
+                };
+                let Ok(()) = svdtools::patch::process_file(
+                    &includer_path,
+                    Some(&patched_svd_path),
+                    None,
+                    &Default::default(),
+                ) else {
+                    println!("cargo::error=Failed to patch svd for {}!", mcu);
+                    return Err(());
+                };
+
+                svd_content = fs::read_to_string(&patched_svd_path);
+                map.insert(mcu.to_owned(), patched_svd_path);
+            }
+            Ok(false) => {
+                // No patching needed, just continue with unpatched.
+                svd_content = fs::read_to_string(&unpatched_svd_path);
+                map.insert(mcu.to_owned(), unpatched_svd_path);
+            }
+            Err(e) => {
+                println!(
+                    "cargo::error=Failed to check patch existence for {}!\n{}",
+                    mcu, e
+                );
+                return Err(());
+            }
+        }
+        let svd_content = match svd_content {
+            Ok(c) => c,
+            Err(e) => {
+                println!(
+                    "cargo::error=Failed to read patched SVD back for {}!\n{}",
+                    mcu, e
+                );
+                return Err(());
+            }
+        };
+        // Apply svd2rust.
+        let mut svd2rust_config = svd2rust::Config::default();
+        svd2rust_config.target = svd2rust::Target::None;
+        svd2rust_config.generic_mod = true;
+        svd2rust_config.make_mod = true;
+        svd2rust_config.strict = true;
+        svd2rust_config.output_dir = Some(gen_dirs.module.clone());
+        svd2rust_config.skip_crate_attributes = true;
+        let generated = match svd2rust::generate(&svd_content, &svd2rust_config) {
+            Ok(g) => g,
+            Err(e) => {
+                println!(
+                    "cargo::error=Failed to generate rust code for {}!\n{}",
+                    mcu, e
+                );
+                return Err(());
+            }
+        };
+
+        let mut syntax_tree: syn::File = match syn::parse_str(&generated.lib_rs) {
+            Ok(s) => s,
+            Err(e) => {
+                println!("cargo::error=Failed to parse rust code for {}! {}", mcu, e);
+                return Err(());
+            }
+        };
+        syntax_tree.attrs.clear();
```

Comment:
This line is superfluous with `skip_crate_attributes` above, but since the PR will transition to older svd2rust that doesn't have it, I left it here for now.